controllers/user/userController.js

```diff
@@ -31,7 +35,10 @@ export const createUserController = async (req, res) => {
       });
     });
 
-    res.json(newUser);
+    // Generate JWT token with the role
+    const token = jwt.sign({ role: 'user', user_id: user_id }, 'your_secret_key');
```

Comment:
move the secret key here to env and let us know through Whatsapp what's the key.
make a good password for the secret key ya.